src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/Hdced4Calculator.kt

```diff
@@ -49,9 +49,9 @@ class Hdced4Calculator(val hdcedConfiguration: Hdced4Configuration) {
       .minus(sentenceCalculation.calculatedUnusedReleaseAda)
 
     // Now that we have sentences > 4 years there are rules around how Consecutive sentences containing SDS+ are calculated
-    if (sentence is ConsecutiveSentence && sentence.isMadeUpOfSdsAndSdsPlusSentences() && nonSdsPlusGreaterThanMinimum(sentence) && nonSdsLessThanMidpoint(sentence) && !lastSentenceIsSdsPlus(sentence)) {
-      val latestNcrd = sentence.orderedSentences.filter { it.offence.isPcscSdsPlus || it.offence.isPcscSds }.maxBy { it.sentenceCalculation.releaseDate }
-      val nonSdsPlusSentences = sentence.orderedSentences.filter { !it.offence.isPcscSdsPlus || !it.offence.isPcscSds }
+    if (sentence is ConsecutiveSentence && sentence.isSdsPlus() && nonSdsPlusGreaterThanMinimum(sentence) && nonSdsLessThanMidpoint(sentence) && !lastSentenceIsSdsPlus(sentence)) {
```

Comment:
Haven't you negated doing the rest of this if by checking that all the sentences are SDS+?

Reply:
Yes - I was doing some playing around with the code, and forgot to remove this. Great spot.